python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -846,38 +851,77 @@ def get_updated_parent_asset_partitions(
             )
         return updated_parents
 
-    @cached_method
     def get_root_unreconciled_ancestors(
-        self, *, asset_partition: AssetKeyPartitionKey, respect_materialization_data_versions: bool
+        self, *, asset_partition: AssetKeyPartitionKey
     ) -> AbstractSet[AssetKey]:
         """Return the set of root unreconciled ancestors of the given asset partition, i.e. the set
         of ancestors of this asset partition whose parents have been updated more recently than
         they have.
         """
-        if self.asset_graph.is_source(asset_partition.asset_key):
-            return set()
+        if asset_partition in self._root_unreconciled_ancestors_cache:
+            return self._root_unreconciled_ancestors_cache[asset_partition]
 
-        parent_asset_partitions = self.asset_graph.get_parents_partitions(
-            dynamic_partitions_store=self,
-            current_time=self._evaluation_time,
-            asset_key=asset_partition.asset_key,
-            partition_key=asset_partition.partition_key,
-        ).parent_partitions
-
-        updated_parents = self.get_updated_parent_asset_partitions(
-            asset_partition,
-            parent_asset_partitions,
-            respect_materialization_data_versions=respect_materialization_data_versions,
-        )
+        # First gather all the candidates that weren't previously computed as well as their parents
```

Comment:
Are we actually including the parents? It looks like we're filtering them out before adding them to visited if they were previously computed?

Reply:
updated language